### PR TITLE
CC3XX: Add AEAD single-part entry points

### PR DESCRIPTION
**Pull request by Vge0rge**
- AEAD enabled in wrapper

Signed-off-by: Georgios Vasilakis <georgios.vasilakis@nordicsemi.no>

### CODE REVIEW

File: include/psa/crypto_driver_contexts_primitives.h [hunk 1]
```diff
@@ -40,6 +40,10 @@
 #if defined(PSA_CRYPTO_DRIVER_CC3XX)
 #include "cc3xx_crypto_primitives.h"
 #endif /* PSA_CRYPTO_DRIVER_CC3XX */
+/* Change the path to fit your machine */
+#include "<path>/cc3xx_psa_aead.h"
+#include "<path>/cc3xx_psa_hash.h"
+#endif
```
Comment (noonfom) [inline include/psa/crypto_driver_contexts_primitives.h:46]: Could you move these includes into `psa_crypto_driver_wrappers.c` below the [`cc3xx.h`](https://github.com/noonfom/mbedtls/blob/4fca8781aebd2051f16d02f22a7acb376891a967/library/psa_crypto_driver_wrappers.c#L53) include. Also, the dummy `<path>` is not required as you defined these as PUBLIC in the [CMake](https://git.trustedfirmware.org/TF-M%2Ftrusted-firmware-m.git/tree/lib%2Fext%2Fcryptocell-312-runtime%2Fcodesafe%2Fsrc%2Fpsa_driver_api%2FCMakeLists.txt?h=refs%2Fchanges%2F34%2F11034%2F2#n9).

Personally, I would prefer if you added the includes to `cc3xx.h` from @salomethirot-arm's [patch](https://review.trustedfirmware.org/c/TF-M/trusted-firmware-m/+/10607). That way only a single include is required here, i.e. `#include "cc3xx.h"`.

**Review by noonfom — CHANGES_REQUESTED** [review]

File: include/psa/crypto_driver_contexts_primitives.h (hunk 1 above)
Comment (noonfom) [reply include/psa/crypto_driver_contexts_primitives.h:46]: Also, that's a stray `#endif` on line 46

File: library/psa_crypto_driver_wrappers.c [hunk 2]
```diff
@@ -1332,7 +1332,7 @@ psa_status_t psa_driver_wrapper_hash_setup(
 #if defined(PSA_CRYPTO_DRIVER_CC3XX)
     status = cc3xx_hash_setup(&operation->ctx.cc3xx_driver_ctx, alg);
     if( status == PSA_SUCCESS )
-        operation->id = PSA_CRYPTO_CC3XX_DRIVER_ID;
+        operation->id = PSA_CRYPTO_DRIVER_CC3XX_ID;
```
Comment (noonfom) [inline library/psa_crypto_driver_wrappers.c:1335]:
```suggestion
        operation->id = PSA_CRYPTO_CC3XX_DRIVER_ID;
```

The original one is more in line with the Mbed TLS convention. Besides, the second macro is undeclared at this point.

**Comment by noonfom** [comment]
@Vge0rge please rebase onto `feature-cc-psa-crypto-drivers`

File: include/psa/crypto_driver_contexts_primitives.h (hunk 1 above)
Comment (Vge0rge) [reply include/psa/crypto_driver_contexts_primitives.h]: Good idea, I removed the includes and I will open a PR to add these in the cc3xx.h file

File: library/psa_crypto_driver_wrappers.c (hunk 2 above)
Comment (Vge0rge) [reply library/psa_crypto_driver_wrappers.c]: Not there anymore.

File: library/psa_crypto_driver_wrappers.c [hunk 3]
```diff
@@ -1537,6 +1537,18 @@ psa_status_t psa_driver_wrapper_aead_encrypt(
             if( status != PSA_ERROR_NOT_SUPPORTED )
                 return( status );
 #endif /* PSA_CRYPTO_DRIVER_TEST */
+#if defined(PSA_CRYPTO_DRIVER_CC3XX)
+            status = cc3xx_psa_aead_encrypt(
```
Comment (noonfom) [inline library/psa_crypto_driver_wrappers.c:1541]:
```suggestion
            status = cc3xx_aead_encrypt(
```

File: library/psa_crypto_driver_wrappers.c [hunk 4]
```diff
@@ -1589,6 +1601,18 @@ psa_status_t psa_driver_wrapper_aead_decrypt(
             if( status != PSA_ERROR_NOT_SUPPORTED )
                 return( status );
 #endif /* PSA_CRYPTO_DRIVER_TEST */
+#if defined(PSA_CRYPTO_DRIVER_CC3XX)
+            status = cc3xx_psa_aead_decrypt(
```
Comment (noonfom) [inline library/psa_crypto_driver_wrappers.c:1605]:
```suggestion
            status = cc3xx_aead_decrypt(
```

**Review by noonfom — CHANGES_REQUESTED** [review]
A few more changes are required since latest [patchset](https://review.trustedfirmware.org/c/TF-M/trusted-firmware-m/+/11034) on TF-M.

File: library/psa_crypto_driver_wrappers.c (hunk 3 above)
Comment (Vge0rge) [reply library/psa_crypto_driver_wrappers.c]: Should be done

File: library/psa_crypto_driver_wrappers.c (hunk 4 above)
Comment (Vge0rge) [reply library/psa_crypto_driver_wrappers.c]: Should be done

**Review by adeaarm — APPROVED** [review]
LGTM

**Review by noonfom — APPROVED** [review]

**Pull request merged**